beacon-chain/core/validators/validator.go

```diff
@@ -383,6 +404,21 @@ func CheckValidatorMinDeposit(validatorSet []*pb.ValidatorRecord, currentSlot ui
 	return validatorSet
 }
 
+// EffectiveBalance returns the balance at stake for the validator.
```

Comment:
Can you add some context to this? Is this because we might allow validators to top off their balance above MAX_DEPOSIT, but they can be slashed at most MAX_DEPOSIT at any time? So we consider their balance at stake to be the lesser of their actual balance or MAX_DEPOSIT.

Reply:
That's correct. A validator can top off. For example I can top off to 45ETH but only 32ETH will be at play. The 7ETH on top is just buffer. I'll update the comments

Reply:
Thanks!